### PR TITLE
Missing sync, missing interval processing, todo review

### PR DESCRIPTION
**Pull request by paulb777**
- Add missing sync from the Objective C port
- Add missing interval adjustment when retry times change from the Objective C port
- Review all TODOs in FriebaseStorage. See review notes for more detail

#no-changelog

### CODE REVIEW

File: FirebaseStorage/Sources/AsyncAwait.swift [hunk 1]
```diff
@@ -30,7 +30,6 @@ import Foundation
     /// - Returns: Data object.
     func data(maxSize: Int64) async throws -> Data {
       return try await withCheckedThrowingContinuation { continuation in
-        // TODO: Use task to handle progress and cancellation.
```
Comment (paulb777) [inline FirebaseStorage/Sources/AsyncAwait.swift:33]: New plan is to address with new v2 APIs and tests. See #10161

File: FirebaseStorage/Sources/Internal/StorageTokenAuthorizer.swift [hunk 2]
```diff
@@ -64,7 +64,6 @@ internal class StorageTokenAuthorizer: NSObject, GTMSessionFetcherAuthorizer {
         request?.setValue(tokenResult.token, forHTTPHeaderField: "X-Firebase-AppCheck")
 
         if let error = tokenResult.error {
-          // TODO: Define better way to use FIRLogger from Swift.
```
Comment (paulb777) [inline FirebaseStorage/Sources/Internal/StorageTokenAuthorizer.swift:67]: This solution is probably fine until we do a Swift logging redesign.

File: FirebaseStorage/Sources/Storage.swift [hunk 3]
```diff
@@ -97,19 +97,31 @@ import FirebaseAuthInterop
    * The maximum time in seconds to retry an upload if a failure occurs.
    * Defaults to 10 minutes (600 seconds).
    */
-  @objc public var maxUploadRetryTime: TimeInterval
```
Comment (paulb777) [inline FirebaseStorage/Sources/Storage.swift:100]: I missed this adjustment in the first port. The Objective C implementation was also synchronized on `self`, but I'm not sure how to do that in Swift or if it's needed.

File: FirebaseStorage/Sources/Storage.swift [hunk 4]
```diff
@@ -291,9 +303,10 @@ import FirebaseAuthInterop
 
   /// Map of apps to a dictionary of buckets to GTMSessionFetcherService.
   private static var fetcherServiceMap: [String: [String: GTMSessionFetcherService]] = [:]
-  private static var retryWhenOffline: GTMSessionFetcherRetryBlock = { (suggestedWillRetry: Bool,
-                                                                        error: Error?,
-                                                                        response: @escaping GTMSessionFetcherRetryResponse) in
+  private static var retryWhenOffline: GTMSessionFetcherRetryBlock = {
```
Comment (paulb777) [inline FirebaseStorage/Sources/Storage.swift:306]: For some reason swiftformat doesn't get this >100 line length.

File: FirebaseStorage/Sources/Storage.swift [hunk 5]
```diff
@@ -308,6 +321,8 @@ import FirebaseAuthInterop
                                                _ auth: AuthInterop,
                                                _ appCheck: AppCheckInterop)
     -> GTMSessionFetcherService {
+    objc_sync_enter(fetcherServiceMap)
```
Comment (paulb777) [inline FirebaseStorage/Sources/Storage.swift:324]: Missed in the port.

File: FirebaseStorage/Sources/StorageComponent.swift [hunk 6]
```diff
@@ -23,9 +23,6 @@ import FirebaseCore
 @objc(FIRStorageProvider)
 protocol StorageProvider {
   @objc func storage(for bucket: String) -> Storage
-  // TODO: See if we can avoid the `type` parameter by either making it a `Storage` argument to
```
Comment (paulb777) [inline FirebaseStorage/Sources/StorageComponent.swift:26]: Comment was copied from FunctionComponent.swift, but not ever relevant for Storage.

File: FirebaseStorage/Sources/StorageUploadTask.swift [hunk 7]
```diff
@@ -39,7 +39,7 @@ import Foundation
     weak var weakSelf = self
     DispatchQueue.global(qos: .background).async {
       guard let strongSelf = weakSelf else { return }
-      if let contentValidationError = strongSelf.isContentToUploadInvalid() {
+      if let contentValidationError = strongSelf.contentUploadError() {
```
Comment (paulb777) [inline FirebaseStorage/Sources/StorageUploadTask.swift:42]: Better name

File: FirebaseStorage/Sources/StorageUploadTask.swift [hunk 8]
```diff
@@ -225,8 +225,7 @@ import Foundation
     self.uploadFetcher?.stopFetching()
   }
 
-  internal func isContentToUploadInvalid() -> NSError? {
-    // TODO: - Does checkResourceIsReachableAndReturnError need to be ported here?
```
Comment (paulb777) [inline FirebaseStorage/Sources/StorageUploadTask.swift:229]: No. The code below is checking.

File: FirebaseStorage/Tests/Integration/StorageAsyncAwait.swift [hunk 9]
```diff
@@ -42,7 +42,6 @@ import XCTest
                      metadata.customMetadata!["shinkansen"])
     }
 
-    // TODO: update this test to use Swift error codes.
```
Comment (paulb777) [inline FirebaseStorage/Tests/Integration/StorageAsyncAwait.swift:45]: Separate tests will be written for Swift v2 API

File: FirebaseStorage/Tests/Unit/StorageTests.swift [hunk 10]
```diff
@@ -38,13 +38,16 @@ class StorageTests: XCTestCase {
     _ = storage.reference(forURL: "gs://benwu-test2.storage.firebase.com/child")
   }
 
-  // TODO: Add fatalError unit testing or update API
-  func SKIPtestBucketEnforced() throws {
+  func testBucketEnforced() throws {
```
Comment (paulb777) [inline FirebaseStorage/Tests/Unit/StorageTests.swift:41]: Can do these tests now that we have a Swift throwing API

File: FirebaseStorage/Tests/Unit/StorageTests.swift [hunk 11]
```diff
@@ -239,6 +247,15 @@ class StorageTests: XCTestCase {
     XCTAssertEqual(32.0, Storage.computeRetryInterval(fromRetryTime: 60.0))
   }
 
+  func testRetryTimeChange() throws {
```
Comment (paulb777) [inline FirebaseStorage/Tests/Unit/StorageTests.swift:250]: Add the always missing test for implicit interval changes based on retrytime changes.

**Review by ncooke3 — APPROVED** [review]
LGTM. Thanks for adding your own review comments throughout!

**Pull request merged**